endpoint/endpoint.go

```diff
@@ -251,6 +251,12 @@ func NewEndpointWithTTL(dnsName, recordType string, ttl TTL, targets ...string)
 	}
 }
 
+// WithLabel attaches a label key/value pair to the Endpoint and returns the Endpoint.
+func (e *Endpoint) WithLabel(key, value string) *Endpoint {
```

Comment:
This seems only needed for tests, please use `export_test.go` to have it only available in tests.